celements-xwiki-core/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -6162,7 +6031,7 @@ public boolean hasMinorEdit(XWikiContext context) {
    *          maybe will be useful
    */
   public boolean hasRecycleBin(XWikiContext context) {
-    return "1".equals(Param("xwiki.recyclebin", "1"));
+    return (recycleBinStore != null);
```

Comment:
make this configurable. "RecycleBin" is a feature. Thus "no RecycleBin" is not a fallback for instantiation or similar errors. It must be configurable to verify the system state is correct after initialization.